source/module_md/run_md.cpp

```diff
@@ -99,7 +99,22 @@ void md_line(UnitCell& unit_in, ModuleESolver::ESolver* p_esolver, MD_para& md_p
             unit_in.update_vel(mdrun->vel);
             std::stringstream file;
             file << GlobalV::global_stru_dir << "STRU_MD_" << mdrun->step_ + mdrun->step_rst_;
-            unit_in.print_stru_file(file.str(), 1, 1);
+            // changelog 20240509
+            // because I move out the dependence on GlobalV from UnitCell::print_stru_file
+            // so its parameter is calculated here
+            bool need_orb = GlobalV::BASIS_TYPE=="pw";
+            need_orb = need_orb && GlobalV::psi_initializer;
```

Comment:
Is it a way to avoid using GlobalV?

Reply:
yes I refactored two functions defined in UnitCell, and I find it is called in run_md. I find in run_md there are other places GlobalV is used, so I stop moving out GlobalV there. But inside those functions called, there are no GlobalV variables anymore.